src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1129,7 +1129,16 @@ public abstract static class AnyInstanceVariableNode extends PrimitiveArrayArgum
 
         @Specialization(limit = "getDynamicObjectCacheLimit()")
         protected boolean any(RubyDynamicObject self,
-                @CachedLibrary("self") DynamicObjectLibrary objectLibrary) {
+                @CachedLibrary("self") DynamicObjectLibrary objectLibrary,
+                @Cached BranchProfile hasPropertiesProfile) {
+            var shape = objectLibrary.getShape(self);
```

Comment:
Cool, I didn't know there was DynamicObjectLibrary#getShape(), that's convenient here.